### PR TITLE
[Ability] Fully Implement Synchronize 

### PR DESCRIPTION
**Pull request by frutescens**
## What are the changes the user will see?
Synchronize and Psycho Shift will have the correct interaction, and Synchronize will no longer need a partial/edge case tag. 

## Why am I making these changes?
Shareholder Value up. 

## What are the changes from a developer perspective?
See Bulbapedia Reference: https://bulbapedia.bulbagarden.net/wiki/Psycho_Shift_(move) 
Psycho Shift now applies the status to the target, activating the target's synchronize ability, and then cures the user. 
In addition, while testing, I noticed that if a Poison or Steel Pokemon with Corrosion poisons a Pokemon with Synchronize, Synchronize is unable to poison the attacking poison or steel pokemon back. This removes one edge case condition from Corrosion.

Tests now have been added for Corrosion and Psycho Shift. 

### Screenshots/Videos

https://github.com/user-attachments/assets/327da2fb-6471-4daf-ab5c-74a26f1f3222

## How to test the changes?
1. Have a Pokemon without Synchronize have a status and Psycho Shift in its move pool 
2. Have it use Psycho Shift against a Pokemon with Synchronize
3. The Psycho Shift user should not be poisoned again by the Pokemon with Synchronize b/c its status ailment does not clear until after the move effect phase completes. 

## Checklist
- [x] **I'm using `beta` as my base branch**
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I considered writing automated tests for the issue?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/data/status-effect.ts [hunk 1]
```diff
@@ -9,11 +9,14 @@ export class Status {
   /** Toxic damage is `1/16 max HP * toxicTurnCount` */
   public toxicTurnCount: number = 0;
   public sleepTurnsRemaining?: number;
+  /** Used in situations where a user should heal after an action is taken */
+  public forceHeal: boolean; //
 
-  constructor(effect: StatusEffect, toxicTurnCount: number = 0, sleepTurnsRemaining?: number) {
+  constructor(effect: StatusEffect, toxicTurnCount: number = 0, sleepTurnsRemaining?: number, forceHeal?: boolean) {
```
Comment (DayKev) [inline src/data/status-effect.ts:15]:
```suggestion
  constructor(effect: StatusEffect, toxicTurnCount: number = 0, sleepTurnsRemaining?: number, forceHeal: boolean = false) {
```

File: src/data/status-effect.ts [hunk 2]
```diff
@@ -9,11 +9,14 @@ export class Status {
...
     this.effect = effect;
     this.toxicTurnCount = toxicTurnCount;
     this.sleepTurnsRemaining = sleepTurnsRemaining;
+    this.forceHeal = forceHeal ?? false;
```
Comment (DayKev) [inline src/data/status-effect.ts:19]:
```suggestion
    this.forceHeal = forceHeal;
```

File: src/test/abilities/synchronize.test.ts [hunk 3]
```diff
@@ -93,17 +93,5 @@ describe("Abilities - Synchronize", () => {
...
-    await game.classicMode.startBattle();
-
-    game.move.select(Moves.PSYCHO_SHIFT);
-    await game.phaseInterceptor.to("BerryPhase");
-
-    expect(game.scene.getPlayerPokemon()!.status?.effect).toBe(StatusEffect.PARALYSIS); // keeping old gen < V impl for now since it's buggy otherwise
-    expect(game.scene.getEnemyPokemon()!.status?.effect).toBe(StatusEffect.PARALYSIS);
-    expect(game.phaseInterceptor.log).toContain("ShowAbilityPhase");
-  });
+  }, 20000);
```
Comment (DayKev) [inline src/test/abilities/synchronize.test.ts:96]:
```suggestion
  });
```
Can you go through and remove the leftover `, 20000` from the test file too while you're there please?

File: src/test/abilities/corrosion.test.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    const playerPokemon = game.scene.getPlayerPokemon();
+    const enemyPokemon = game.scene.getEnemyPokemon();
+    expect(playerPokemon!.status).toBeUndefined();
+
+    game.move.select(Moves.SPLASH);
+    await game.phaseInterceptor.to("BerryPhase");
+    expect(playerPokemon!.status).toBeDefined();
+    expect(enemyPokemon!.status).toBeUndefined();
+
+    expect(true).toBe(true);
```
Comment (DayKev) [inline src/test/abilities/corrosion.test.ts:46]:
```suggestion
```

File: src/data/move.ts [hunk 5]
```diff
@@ -2266,24 +2266,23 @@ export class PsychoShiftEffectAttr extends MoveEffectAttr {
     super(false, { trigger: MoveEffectTrigger.HIT });
   }
 
-  apply(user: Pokemon, target: Pokemon, move: Move, args: any[]): boolean {
```
Comment (Tempo-anon) [inline src/data/move.ts:2269]: Since you're here in the `PsychoShiftEffectAttr`, you should add some documentation. Also, is there a reason you removed the boolean return type hint for this apply?

Comment (frutescens) [reply src/data/move.ts:2269]: oh my bad,

**Comment by DayKev** [comment]
Sorry to mess with the git history but since Temp doesn't pay attention to the commit messages when he merges PRs I gotta make sure. :pikastare:

**Review by DayKev — APPROVED** [review]

**Review by innerthunder — APPROVED** [review]

**Comment by Tempo-anon** [comment]
> Sorry to mess with the git history but since Temp doesn't pay attention to the commit messages when he merges PRs I gotta make sure. :pikastare: image

Posting here as well.

I'm okay with some more silliness in the git history as long as there's no identifying information or offensive content since nobody really reads specific commit messages in a squashed commit. A squashed commit's title should be informative and correct though, and that's what I care more about.
